### DISCUSSION TITLE
Client has problem to access Doris behind NAT

### DISCUSSION THREAD

**Discussion by ziudu** [apache/doris#23655] [RFC]
Dear, 

I've a configuration problem:
Doris cluster is deployed in a private network. Its ports could be mapped to an outside address via NAT. Topology:
---------------------Topology-------------------------------------- 
client --> Gateway(10.10.10.10) --> Doris Cluster(192.168.0.1/2/3)
---------------------------------------------------------------------
However, the client report an error with internal IP: 
Caused by: org.apache.doris.spark.exception.StreamLoadException: stream load error: http request exception,load url : http://192.168.0.1:8040/api/xxx/xxx/_stream_load?,failed to execute spark streamload with label: spark_streamload_20230830_135909_c71dcc3fe3354687841fbe87f325e85b

Is NAT configuration supported by Doris？If yes，how should I do?

**Comment by ziudu** [comment]
it seems during the stream load process, the backend private IP address was returned incorrectly to the client

**Reply by LemonLiTree** [reply]
Has this issue been resolved?

**Comment by ziudu** [comment]
We believe setting a proxy before BE could solve this problem, as is described in the documentation. 
But it is not done yet due to delays in network policy configuration changes...
Thanks. 
















At 2023-09-01 16:57:44, "Liqf" ***@***.***> wrote:

Has this issue been resolved?
